.circleci/config.base.yml

```diff
@@ -514,6 +516,38 @@ jobs:
           path: ~/repo/packages/amplify-migration-tests/amplify-migration-reports
     working_directory: ~/repo
 
+  amplify_migration_tests_v10:
+    parameters:
+      os:
+        type: executor
+        default: l_large
+    executor: << parameters.os >>
+    environment:
+      AMPLIFY_PATH: /home/circleci/.amplify/bin/amplify
+    steps:
+      - restore_cache:
+          key: amplify-cli-repo-{{ .Branch }}-{{ .Revision }}
+      - restore_cache:
+          key: amplify-cli-yarn-deps-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - restore_cache:
+          key: amplify-pkg-binaries-{{ .Branch }}-{{ .Revision }}
+      - run:
+          name: Run tests migrating from CLI v10.5.1
+          command: |
+            source .circleci/local_publish_helpers.sh
+            changeNpmGlobalPath
+            cd packages/amplify-migration-tests
+            unset IS_AMPLIFY_CI
+            echo $IS_AMPLIFY_CI
```

Comment:
this is a small hack, unfortunately, in v10 there is custom logic that checks for this parameter, and will try to install the wrong version using a commit hash. That feature interferes with us trying to test a specific version, such as 10.5.1, but this was not an issue in v6/v5
In order for the script to download the exact version, we must disable this "commit hash" feature manually.

![image](https://user-images.githubusercontent.com/110861985/204679595-c858bf43-8147-4df5-bf5e-1eabaeabd5d0.png)

Reply:
I would recommend to do this:
```
  if (process.env.IS_AMPLIFY_CI && process.env.IS_AMPLIFY_CI_USE_LATEST_BINARY) { // exact naming of extra env var tbd.
    url = url.replace('.tgz', `-${getCommitHash()}.tgz`);
  }
```
or
```
  if (process.env.IS_AMPLIFY_CI && process.env.BINARY_URL_OVERRIDE) {
    url = BINARY_URL_OVERRIDE;
  }
```

using just `IS_AMPLIFY_CI` to work around this is going to be problematic in the future if there is another logic that would like to use that to branch out.

Reply:
fwiw I like the `BINARY_URL_OVERRIDE` more. and I'd shift this getComitHash logic to the caller (script).

Reply:
we can't modify this in 10.5.1 or previous versions; the logic executes from the existing/old release as it's already part of the binary files; if we change this now in v10.5.2/v10.6 we will still need this custom logic if we add migration tests for v9 or v10.4

Reply:
gotcha. good to go as is in this PR. I'd log follow up task to get this straight for future releases.

we can keep IS_AMPLIFY_CI as is. I'd just add extra knob to control this particular piece of code and use it in migration tests, i.e. `IS_AMPLIFY_CI && URL_SPECIFIC_VAR`